templates/template.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require "erb"
+require "fileutils"
+require "yaml"
+
+module RBS
+  class Template
+    class Field
+      attr_reader :name
+      attr_reader :c_name
+      attr_reader :c_type
+
+      def initialize(yaml)
+        @name = yaml["name"]
+        @c_name = yaml["c_name"] || @name
+        @c_type = "VALUE"
+      end
+    end
+
+    class Type
+      attr_reader :full_name
+      attr_reader :name
+      attr_reader :c_function_name
+      attr_reader :c_constant_name
+      attr_reader :parent_c_constant_name
+      attr_reader :fields
+
+      def initialize(yaml)
+        @full_name = yaml["name"]
+        @name = yaml["name"].split("::").last
+        @c_function_name = if @full_name =~ /^RBS::Types/
+          name = @full_name.gsub("::", "_").underscore
+          name.gsub("_types_", "_") # TODO: should we rename these methods?
```

Comment:
Let's change this TODO to explain that we're going to do this next and remove this whole `if` statement